src/core/qgsvectorlayer.cpp

```diff
@@ -3923,6 +3918,15 @@ QVariant QgsVectorLayer::aggregate( QgsAggregateCalculator::Aggregate aggregate,
     return QVariant();
   }
 
+  if ( context )
```

Comment:
Does that block do something?

Reply:
As soon as filter stacking is added,this part will be needed. This is a precursor.

Reply:
Remove any precursors

Reply:
If the stacking is merge before this PR I'll atler it, otherwise this will help me find back the place where to add a few line.

Reply:
@m-kuhn I was wrong its not a precursor anymore. But I'm not sure if my implementation of the pointer is optimal or if it can cause segfault.